extensions/agroal/deployment/src/main/resources/dev-ui/qwc-agroal-datasource.js

```diff
@@ -0,0 +1,700 @@
+                        acc.set(obj.columnName, obj);
+                        return acc;
+                    }, new Map());
+                    this._executeCurrentSQL();
+                }else {
+                    this._displaymessage = "No tables found";
+                }
+            });
+        }
+    }
+    
+    _createImportSql(){
+        if(this._selectedDataSource){
+            this.jsonRpc.getInsertScript({datasource:this._selectedDataSource.name}).then(jsonRpcResponse => {
+                this._insertSQL = jsonRpcResponse.result;
+                this._dialogOpened = true;
+            });
+        }
+    }
+    
+    _executeClicked(){
+        let newValue = this.shadowRoot.getElementById('sql').getAttribute('value');
+        this._executeSQL(newValue);
+    }
+    
+    _clearSqlInput(){
+        if(this._selectedTable){
+            if(this._appendSql){
+                this._executeSQL("select * from " + this._selectedTable.tableName + " " + this._appendSql);
+            }else{
+                this._executeSQL("select * from " + this._selectedTable.tableName);
+            }
+        }
+    }
+    
+    _shiftEnterPressed(event){
+        this._executeSQL(event.detail.content);
+    }
+    
+    _executeSQL(sql){
+        this._currentSQL = sql.trim();
+        this._executeCurrentSQL();
+    }
+    
+    _startsWithIgnoreCaseAndSpaces(str, searchString) {
+        return str.trim().toLowerCase().startsWith(searchString.toLowerCase());
+    }
+    
+    hotReload(){
+        this._fetchTableDefinitions();
+    }
+    
+    _isAllowedHostDatabase() {
+        
+        let jdbcUrl = this._selectedDataSource.jdbcUrl;
+        try {
+            if (jdbcUrl.startsWith("jdbc:h2:mem:") || jdbcUrl.startsWith("jdbc:h2:file:")) {
+                return true;
+            }
+
+            if (jdbcUrl.startsWith("jdbc:derby:memory:")) {
+                return true;
+            }
+
+            if (jdbcUrl.startsWith("jdbc:derby:")) {
+                const derbyUri = jdbcUrl.replace("jdbc:", "");
+                if (derbyUri.startsWith("localhost") || derbyUri.startsWith("127.0.0.1")) {
+                    return true;
+                }
+                if(this._allowedHost && this._allowedHost!="" && derbyUri.startsWith(this._allowedHost)){
+                    return true;
+                }
+            }
+
+            const urlPattern = /^jdbc:[^:]+:\/\/([^:/]+)(:\d+)?/;
+            const match = jdbcUrl.match(urlPattern);
+
+            if (match) {
+                const host = match[1];
+                if(host === "localhost" || host === "127.0.0.1" || host === "::1"){
+                    return true;
+                }
+                if(this._allowedHost && this._allowedHost!="" && host === this._allowedHost){
+                    return true;
+                }
+            }
+
+            return false;
+        } catch (e) {
+            console.error(e);
+            return false;
+        }
+    }
```

Comment:
So... we're essentially doing client-side security? 

Shouldn't this rather be checked server-side in `DatabaseInspector`?

Reply:
It is also on the server side